### ISSUE TITLE
OStorageException: Invalid value of TX counter (Exception during commit of active transaction)

### ISSUE THREAD

**Issue by vtliem** [orientechnologies/orientdb#6928]
## OrientDB Version, operating system, or hardware.
- v2.2.12

## Operating System
-  Linux

## Steps to reproduce the problem

I'm not sure if it's a bug because ` java.lang.OutOfMemoryError: GC overhead limit exceeded` raised.
It's a chat server. When I send many messages in short time (600~1000 in about 10minutes). This exception raised. After that, any time I save a record, ODatabaseException will raise.

[orientdb-0_0.log.txt](https://github.com/orientechnologies/orientdb/files/598813/orientdb-0_0.log.txt)
-----------------------------------
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Error in data flush background thread, please restart database and send full stack trace inside of bug report
	DB name="main"
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.checkLowDiskSpaceFullCheckpointRequestsAndBackgroundDataFlushExceptions(OAbstractPaginatedStorage.java:4185)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.createRecord(OAbstractPaginatedStorage.java:1036)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeSaveRecord(ODatabaseDocumentTx.java:2095)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveNew(OTransactionNoTx.java:246)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:179)
	... 28 more
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded

**Comment by laa** [comment]
Hi @vtliem that is definitively caused by out of memory exception. Could you raise amount of memory, but before that could you make import/export procedure to fix possible issues with database ?

**Comment by vtliem** [comment]
I changed from
`-Xmx1000m -Dstorage.diskCache.bufferSize=500`
to `-Xmx2000m -Dstorage.diskCache.bufferSize=500` and can send 1000 message in 10minutes.

I also did import/export to clean up database.

I read OrientDB document and found 

> So rather than:
> `java -Xmx8g ...`
> You could instead try this:
> `java -Xmx800m -Dstorage.diskCache.bufferSize=7200 ...`

but if limit heap too low, `java.lang.OutOfMemoryError: GC overhead limit exceeded` raises if do a big transaction or many write in a short time (db is not too big, ~ 30k records + 500k index entries). In this case, is 500MB of bufferSize enough or I need more?
